client/config.go

```diff
@@ -12,16 +14,33 @@ type Config struct {
 	Keysize                int    `mapstructure:"key_size"`
 	Validity               string `mapstructure:"validity"`
 	ValidateTLSCertificate bool   `mapstructure:"validate_tls_certificate"`
+	PublicFilePrefix       string `mapstructure:"public_file_prefix"`
 }
 
 func setDefaults() {
 	viper.BindPFlag("ca", pflag.Lookup("ca"))
 	viper.BindPFlag("key_type", pflag.Lookup("key_type"))
 	viper.BindPFlag("key_size", pflag.Lookup("key_size"))
 	viper.BindPFlag("validity", pflag.Lookup("validity"))
+	viper.BindPFlag("public_file_prefix", pflag.Lookup("public_file_prefix"))
 	viper.SetDefault("validateTLSCertificate", true)
 }
 
+// expandTilde expands ~ and ~user for a given path.
+func expandTilde(path string) string {
+	re := regexp.MustCompile("^~([^/]*)(/.*)")
+	if m := re.FindStringSubmatch(path); len(m) > 0 {
+		u, _ := user.Current()
+		if m[1] != "" {
+			u, _ = user.Lookup(m[1])
```

Comment:
You're chucking the error here. What happens if the user doesn't exist? If this is called with `expandTilde("~omg/blah")` and `omg` doesn't exist I think what's going to happen is that `"~omg/blah"` will be returned and later code will write some files in the current directory with the names `~omg/blah.pub` and `~omg/blah-cert.pub`.